ui/apps/platform/src/Containers/Collections/RuleSelector/AutoCompleteSelect.tsx

```diff
@@ -27,25 +40,51 @@ function getOptions(
 
 export function AutoCompleteSelect({
     id,
+    collection,
+    entityType,
+    autocompleteField,
     selectedOption,
     className = '',
     typeAheadAriaLabel,
     onChange,
+    placeholder,
     validated,
     isDisabled,
-    autocompleteProvider,
-    OptionComponent = SelectOption,
 }: AutoCompleteSelectProps) {
     const { isOpen, onToggle, closeSelect } = useSelectToggle();
     const [typeahead, setTypeahead] = useState(selectedOption);
+
     // When isTyping is true, autocomplete results will not be displayed. This prevents
     // a clunky UX where the dropdown results and the user text get out of sync.
     const [isTyping, setIsTyping] = useState(false);
 
+    // We need to wrap this custom SelectOption component in a forward ref
+    // because PatternFly will pass a `ref` to it
+    const OptionComponent = forwardRef(
```

Comment:
Initially I wanted to keep the custom option rendering and autocomplete calls outside of this component, to make it more reusable, but the extra complexity for the (small?) chance of us needing this someplace else didn't feel worth it.

Without the rendering and the autocomplete behavior, there wasn't a ton of behavior built on top of the PatternFly defaults. We can revisit this in the future if we think we will need this elsewhere.

Reply:
Good call.